rmc/pipeline/regional_constraint.py

```diff
@@ -414,12 +413,16 @@ def main(args):
                 overwrite=args.overwrite,
             )
 
-            if args.filter_outliers:
-                logger.info("Removing outlier transcripts...")
-                constraint_transcripts = get_constraint_transcripts(outlier=False)
-                rmc_ht = rmc_ht.filter(
-                    constraint_transcripts.contains(rmc_ht.transcript)
+            if args.filter_outliers or args.filter_to_canonical:
+                # NOTE: RMC search should be run on only canonical transcripts
+                # rather than filtering to canonical transcripts at this step
+                # for compute efficiency
+                keep_transcripts = get_constraint_transcripts(
+                    all_transcripts=not args.filter_outliers,
+                    filter_to_canonical=args.filter_to_canonical,
+                    outlier=not args.filter_outliers,
```

Comment:
Perhaps add a note in the `get_constraint_transcripts` docstring for `outlier` that this only does something if `all_transcripts` is False